app/gui2/src/components/ComponentDocumentation.vue

```diff
@@ -2,33 +2,35 @@
 import DocumentationPanel from '@/components/DocumentationPanel.vue'
 import { injectGraphSelection } from '@/providers/graphSelection'
 import { useGraphStore } from '@/stores/graph'
-import { ref, watchEffect } from 'vue'
+import { computed, ref } from 'vue'
 import type { SuggestionId } from 'ydoc-shared/languageServerTypes/suggestions'
 import { Err, Ok, type Result } from 'ydoc-shared/util/data/result'
 
+const props = defineProps<{ displayedSuggestionId: SuggestionId | null }>()
 const selection = injectGraphSelection()
 const graphStore = useGraphStore()
 
-const displayedDocs = ref<Result<SuggestionId>>()
-
 function docsForSelection() {
   const selected = selection.tryGetSoleSelection()
   if (!selected.ok) return Err('Select a single component to display help')
   const suggestionId = graphStore.db.nodeMainSuggestionId.lookup(selected.value)
   if (suggestionId == null) return Err('No documentation available for selected component')
   return Ok(suggestionId)
 }
-watchEffect(() => (displayedDocs.value = docsForSelection()))
+
+const displayedSuggestionId = computed(() =>
```

Comment:
Let's avoid shadowing a prop with a variable, to keep template references unambiguous.